snapcraft/pluginhandler.py

```diff
@@ -633,6 +662,56 @@ def _clean_migrated_files(snap_files, snap_dirs, directory):
             os.rmdir(os.path.join(directory, snap_dir))
 
 
+def _linktree(source, destination):
+    for root, directories, files in os.walk(source):
+        for entry in itertools.chain(files, directories):
+            source_path = os.path.join(root, entry)
+            relpath = os.path.relpath(source_path, source)
+            destination_path = os.path.join(destination, relpath)
+
+            if os.path.isdir(source_path) and not os.path.islink(source_path):
+                os.makedirs(destination_path, exist_ok=True)
```

Comment:
@kyrofa do we need to create these potential empty dirs? The `else` clause seems will be able to ensure the directories exist